checker/tests/run-pass/def_id_not_unique.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+//
+
+// A test that call two distinct instantiations of the same generic functions in such a way
+// that both calls will use the same DefId, with the result that the second call will use
+// the wrong summary if using a cache that is keyed by DefId.
+
+// This example is highly contrived, but inspired by a larger way more subtle code fragment.
```

Comment:
supernit, but helps my brain parse this:

  // This example is highly contrived, but inspired by a larger, way more subtle, code fragment.